processor/metricstransformprocessor/metrics_transform_processor_test.go

```diff
@@ -0,0 +1,306 @@
+						Name:      in,
+						LabelKeys: labels,
+					},
+				}
+			}
+
+			// process
+			cErr := amp.ConsumeMetrics(
+				context.Background(),
+				pdatautil.MetricsFromMetricsData([]consumerdata.MetricsData{
+					{},
+					md,
+					{
+						Metrics: []*metricspb.Metric{},
+					},
+				}),
+			)
+			assert.Nil(t, cErr)
+
+			// get and check results
+			got := next.AllMetrics()
+			require.Equal(t, 1, len(got))
+			gotMD := pdatautil.MetricsToMetricsData(got[0])
+			require.Equal(t, 3, len(gotMD))
+			assert.EqualValues(t, consumerdata.MetricsData{}, gotMD[0])
+			require.Equal(t, len(test.outMN), len(gotMD[1].Metrics))
+			for idx, out := range gotMD[1].Metrics {
+				// check name
+				assert.Equal(t, test.outMN[idx], out.MetricDescriptor.Name)
+
+				//check labels
+				if test.action == Insert {
+					// check the inserted is correctly updated
+					if out.MetricDescriptor.Name == test.newName {
+						for lidx, l := range out.MetricDescriptor.LabelKeys {
+							assert.Equal(t, test.outLabels[lidx], l.Key)
+						}
+					}
+					// check the original is untouched
+					if out.MetricDescriptor.Name == test.metricName {
+						for lidx, l := range out.MetricDescriptor.LabelKeys {
+							assert.Equal(t, test.inLabels[lidx], l.Key)
+						}
+					}
+				} else {
+					targetName := test.metricName
+					if test.newName != "" {
+						targetName = test.newName
+					}
+
+					if out.MetricDescriptor.Name == targetName {
+						for lidx, l := range out.MetricDescriptor.LabelKeys {
+							assert.Equal(t, test.outLabels[lidx], l.Key)
+						}
+					}
+				}
+			}
+
+			assert.EqualValues(t, consumerdata.MetricsData{Metrics: []*metricspb.Metric{}}, gotMD[2])
+			assert.NoError(t, amp.Shutdown(ctx))
+		})
+	}
+}
+
+func BenchmarkMetricsTransformProcessorRenameMetrics(b *testing.B) {
+	// runs 1000 metrics through a filterprocessor with both include and exclude filters.
+	stressTest := metricsTransformTest{
+		name:       "1000Metrics",
+		metricName: "metric1",
+		action:     "insert",
+		newName:    "newname",
+	}
+
+	for len(stressTest.inMN) < 1000 {
+		stressTest.inMN = append(stressTest.inMN, inMetricNames...)
+	}
+
+	benchmarkTests := append(standardTests, stressTest)
+
+	for _, test := range benchmarkTests {
+		// next stores the results of the filter metric processor
+		next := &etest.SinkMetricsExporter{}
+		cfg := &Config{
+			ProcessorSettings: configmodels.ProcessorSettings{
+				TypeVal: typeStr,
+				NameVal: typeStr,
+			},
+			MetricName: test.metricName,
+			Action:     test.action,
+			NewName:    test.newName,
+		}
+
+		amp, err := newMetricsTransformProcessor(next, cfg)
+		assert.NotNil(b, amp)
+		assert.Nil(b, err)
+
+		md := consumerdata.MetricsData{
+			Metrics: make([]*metricspb.Metric, len(test.inMN)),
+		}
+
+		for idx, in := range test.inMN {
+			md.Metrics[idx] = &metricspb.Metric{
+				MetricDescriptor: &metricspb.MetricDescriptor{
+					Name: in,
+				},
+			}
+		}
+
+		b.Run(test.metricName, func(b *testing.B) {
```

Comment:
Does this work as expected? I'm not sure this is a proper benchmark test, I think you still need to iterate over `for i := 0; i < b.N; i++` in your test - see https://blog.golang.org/subtests

What output does this generate?

Reply:
I think I do see what you are talking about here, but it seems like that the other processors are also written this way, so I am also following the conventions here. e.g. [filter processor benchmark test](https://github.com/open-telemetry/opentelemetry-collector/blob/969a8d4e9aceb9808f06d61df318956d28855ccc/processor/filterprocessor/filter_processor_test.go#L265)

As for the output, I am seeing something like this:
```
BenchmarkMetricsTransformProcessorRenameMetrics/metric_label_update-16  1000000000  0.000010 ns/op  0 B/op  0 allocs/op

```

Reply:
Yea I think the filter processor benchmark is wrong too. These are just for benchmarking and don't currently run as part of CI, so there's no guarantee the existing tests are correct ☹️ 

Can you update this benchmark as per go documentation?

Reply:
Got it! It is updated now